src/Microsoft.PowerShell.Security/security/CatalogCommands.cs

```diff
@@ -95,25 +95,24 @@ protected override void ProcessRecord()
                 {
                     foreach (PathInfo tempPath in SessionState.Path.GetResolvedPSPathFromPSPath(p))
                     {
-                        if (ShouldProcess(tempPath.ProviderPath))
+                        if (ShouldProcess("Including path " + tempPath.ProviderPath, "", ""))
                         {
                             paths.Add(tempPath.ProviderPath);
                         }
                     }
                 }
             }
 
-            // We add 'paths.Count > 0' to support 'ShouldProcess()'
-            if (paths.Count > 0 )
-            {
-                string drive = null;
+            string drive = null;
 
-                // resolve catalog destination Path
-                if (!SessionState.Path.IsPSAbsolute(catalogFilePath, out drive) && !System.IO.Path.IsPathRooted(catalogFilePath))
-                {
-                    catalogFilePath = SessionState.Path.GetUnresolvedProviderPathFromPSPath(catalogFilePath);
-                }
+            // resolve catalog destination Path
+            if (!SessionState.Path.IsPSAbsolute(catalogFilePath, out drive) && !System.IO.Path.IsPathRooted(catalogFilePath))
+            {
+                catalogFilePath = SessionState.Path.GetUnresolvedProviderPathFromPSPath(catalogFilePath);
+            }
 
+            if (ShouldProcess(catalogFilePath))
```

Comment:
Sorry for being late. Above we made `"Including path " + tempPath.ProviderPath` but here we show only path `catalogFilePath` - different style message.

Reply:
`ShouldProcess()` here automatically shows the cmdlet name as the action, so it says something like: What If: Executing Test-FileCatalog against "catalogFilePath"

Reply:
My question is about the difference - if an user set both `Path` and `CatalogFilePath` parameters he can get some requests for _files_ from `Path` and then different style request for catalogFilePath - he can not understand that last request is for just catalog _file_.

Reply:
If both is specified, you see this:

```powershell
PS C:\Users\steve\test> New-FileCatalog -Path C:\Users\steve\test,c:\,c:\Users\steve -CatalogFilePath c:\users\steve\tes
t\test.cat -WhatIf
What if: Including path C:\Users\steve\test
What if: Including path C:\
What if: Including path C:\Users\steve
What if: Performing the operation "New-FileCatalog" on target "c:\users\steve\test\test.cat".
```

Reply:
Looks good. Thanks!